code/clue.js

```diff
@@ -1,64 +1,297 @@
+  return array[randomIndex]
 }
 
-// CREATE OBJECTS FOR ALL THE WEAPONS, ADD MORE CHARACTERISTICS THE THE WEAPONS IF YOU LIKE.
+let mystery = {
+  killer: "",
+  weapon: "",
+  room: ""
+}
+
+const pickKiller = () => {
+  mystery.killer = pickRandom(suspects)
 
-const rope = {
-  name: "rope",
-  weight: 10
+  const theKiller = document.getElementById("killer")
+  const theKillerName = document.getElementById("killerName")
+  const theKillerImage = document.getElementById("killerImage")
+  const theKillerAge = document.getElementById("killerAge")
+  const theKillerOccupation = document.getElementById("killerOccupation")
+  const theKillerDescription = document.getElementById("killerDescription")
+
+  theKiller.style.background = mystery.killer.color
+  theKillerName.innerHTML = mystery.killer.firstName + " " + mystery.killer.lastName
+  theKillerImage.src = mystery.killer.image
+  theKillerAge.innerHTML = mystery.killer.age
+  theKillerOccupation.innerHTML = mystery.killer.occupation
+  theKillerDescription.innerHTML = mystery.killer.description
+
+  hideTextShowButton()
 }
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const pickWeapon = () => {
+  mystery.weapon = pickRandom(weapons)
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS.
+  const theWeapon = document.getElementById("weapon")
+  const theWeaponName = document.getElementById("weaponName")
+  const theWeaponWeight = document.getElementById("weaponWeight")
+  const theWeaponImage = document.getElementById("weaponImage")
 
-const suspects = [
-  mrGreen,
-  mrsWhite,
-  profPlum,
-  missScarlet,
-  mrsPeacock,
-  mrMustard
-]
+  theWeapon.style.background = mystery.weapon.color
+  theWeaponName.innerHTML = mystery.weapon.name
+  theWeaponWeight.innerHTML = mystery.weapon.weight
+  theWeaponImage.src = mystery.weapon.image
 
-const weapons = []
+  hideTextShowButton()
+}
 
-const rooms = []
+const pickRoom = () => {
+  mystery.room = pickRandom(rooms)
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FOR THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
+  const theRoom = document.getElementById("room")
+  const theRoomName = document.getElementById("roomName")
+  const theRoomFloor = document.getElementById("roomFloor")
+  const theRoomImage = document.getElementById("roomImage")
+
+  theRoom.style.background = mystery.room.color
+  theRoomName.innerHTML = mystery.room.name
+  theRoomFloor.innerHTML = mystery.room.floor
+  theRoomImage.src = mystery.room.image
+
+  hideTextShowButton()
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+const revealMystery = () => {
+  document.getElementById("mystery").innerHTML = `The murder was committed by ${mystery.killer.displayName}, in the ${mystery.room.name.toLowerCase()} with the ${mystery.weapon.name.toLowerCase()}.`
```

Comment:
When clicking on the button without clicking on the cards before you got this error `Cannot read property 'toLowerCase' of undefined`. This will simply be solved if you add a if statement to this function, where you look if you have everything you need to write out this sentence